### PR TITLE
Fix/vprotocol initialization

### PR DESCRIPTION
**Pull request by bosilca**
A fix for issue #7247.

### CODE REVIEW

**Comment by ibm-ompi** [comment]
The IBM CI (GNU Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/ibm-ompi/88b480fb98dfd2bdac939f023e1f8fe6

**Comment by ibm-ompi** [comment]
The IBM CI (XL Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/ibm-ompi/88b480fb98dfd2bdac939f023e1f8fe6

File: ompi/mca/vprotocol/pessimist/vprotocol_pessimist_component.c [hunk 1]
```diff
@@ -111,16 +111,22 @@ static int mca_vprotocol_pessimist_component_close(void)
...
 
     /* sanity check */
-    if(enable_mpi_threads)
-    {
+    if(enable_mpi_threads) {
+        /**
+         * Prevent the pessimistic protocol from activating if we are in a potentially multithreaded
+         * applications. The reason is that without tracking the thread initiator of messages it is
+         * extrmely difficult to provide a global message ordering in threaded-scenarios. Thus, the
+         * safe approach is to turn off the logger.
```
Comment (abouteiller) [inline ompi/mca/vprotocol/pessimist/vprotocol_pessimist_component.c]: Maybe we could have an override flag for people that really know what they are doing. It is fine to use MT as long as the matching and post-order remains deterministic per communicator (i.e. a communicator per thread approach would work fine).

**Comment by awlauria** [comment]
bot:ompi:retest

**Comment by awlauria** [comment]
bot:ibm:retest

File: ompi/mca/vprotocol/pessimist/vprotocol_pessimist_component.c (hunk 1 above)
Comment (bosilca) [reply ompi/mca/vprotocol/pessimist/vprotocol_pessimist_component.c]: I have added such support via the ```vprotocol_pessimist_allow_thread_multiple``` MCA parameter.

Comment (abouteiller) [reply ompi/mca/vprotocol/pessimist/vprotocol_pessimist_component.c]: +1

**Comment by bosilca** [comment]
bot:aws:retest

**Pull request merged**